### PR TITLE
Add assertions about RESET message count for optimized drivers

### PR DESCRIPTION
**Pull request by robsdedude**
Amending tests added in https://github.com/neo4j-drivers/testkit/pull/394

### CODE REVIEW

File: tests/stub/connectivity_check/test_get_server_info.py [hunk 1]
```diff
@@ -176,6 +181,7 @@ def test_routing_from_pool(self):
                 self.assertGreater(resets2, resets1)
         self._router.done()
         self._server1.done()
+        self._server1._dump()
```
Comment (bigmontz) [inline tests/stub/connectivity_check/test_get_server_info.py]: is it a leftover?

**Review by bigmontz — APPROVED** [review]
🐎 

(Apart of a small comment)

File: tests/stub/connectivity_check/test_get_server_info.py (hunk 1 above)
Comment (robsdedude) [reply tests/stub/connectivity_check/test_get_server_info.py]:
```suggestion
```

**Comment by robsdedude** [comment]
The failing tests with the Java driver are not related to the change.

**Pull request merged**